### DISCUSSION TITLE
Break down recipe instructions further

### DISCUSSION THREAD

**Discussion by luizribeiro** [cooklang/spec#59] [Ideas]
I think it would be nice to be able to break down recipe's instructions further. This sometimes make long recipes easier to follow. For example:

```cook
# Dough

Mix @flour, @yeast, ...

Knead by hand or beat the dough with your #mixer

....

Roll dough into a 12-inch circle. Cover and rest as you prep the pizza toppings.

# Toppings

Spread @marinara sauce{} over base, followed by @cheese.

...

# Baking

Bake it at very high temperature for ~{15%minutes}.
```

I'm borrowing the syntax for sections from markdown headings, but I do realize there is a bit of a conflict with how we mark cookware right now.

An interesting thought might be to, instead of writing sections, representing the steps for producing a product instead.

For example:

```cook
$dough: {
  Mix @flour, @yeast, ...

  Knead by hand or beat the dough with your #mixer

  ...
}

$toppings: {
  ...
}

Roll $dough into a 12-inch circle. Cover and rest.

Spread @marinara sauce{} over base, followed by @cheese and the prepared $toppings.

Bake it at very high temperature for ~{15%minutes}.
```

The interesting thing of representing it this way is that it would unlock us to ability to reuse and do combinations of different recipes (e.g. different recipes for pizza toppings, while using the same dough recipe). It also helps with readability a lot IMO since it separates clearly the different parts of the recipe by their products (which could even be used later on in other parts of the recipe).

**Comment by dubadub** [comment]
Hi @luizribeiro I'm wondering if including dough and topping recipes from [separate files](https://github.com/cooklang/spec/issues/10) will solve the same issue?

**Comment by DanMacKay1961** [comment]
I think this is a necessary feature too.  I suggest using a meta called >>section:

**Reply by Net-Mist** [reply]
I agree. This is a natural solution I think. See for instance https://raw.githubusercontent.com/Net-Mist/remy/98464bb9868657fd4c6ed12f3d3081a19a91aca5/recipes/plats/Flammekueche.cook

**Comment by PlayerCreed** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by dubadub** [comment]
Relevant proposal for spec change https://github.com/cooklang/spec/pull/104.

**Comment by dubadub** [comment]
This is now part of the main specification, closing the discussion.
